compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeResolver.java

```diff
@@ -1450,7 +1450,7 @@ private void fillEffectiveType(BIntersectionType intersectionType,
         Iterator<BType> iterator = intersectionType.getConstituentTypes().iterator();
```

Comment:
As I understand, the constituent types are used to determine the effective type of the intersection type. If that's the case, we can add the implied type directly to the list of constituent types, without having to perform the `getImpliedType` operation here. Please verify this as well.